SchoolManagerSystem.Service/Courses/Implementations/CourseServices.cs

```diff
@@ -0,0 +1,160 @@
+﻿using SchoolManagerSystem.Common.DTOs;
+using SchoolManagerSystem.Repository.UnitOfWork.Interfaces;
+using SchoolManagerSystem.Service.Courses.Interfaces;
+using SchoolManagerSystem.Service.Validations.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SchoolManagerSystem.Service.Courses.Implementations
+{
+	public class CourseServices : ICourseServices
+	{
+		private readonly IUnitOfWork _unit;
+		private readonly IRequestValidations _validate;
+
+		public CourseServices(IUnitOfWork unit, IRequestValidations validate)
+		{
+			_unit = unit;
+			_validate = validate;
+		}
+
+		public async Task<CourseResponse> AddCourse(CourseRequest request)
+		{
+			if (!_validate.IsCourseNameValid(request.CourseName))
+			{
+				throw new ArgumentException("Course Name Is Invalid");
+			}
+
+			var student = await _unit.Student.FetchStudentAsync(request.LevelId);
+			if (student.Count == 0)
+			{
+				throw new ArgumentNullException($"invalid levelId {request.LevelId} ");
+			}
+
+			var course = await _unit.Course.AddCourse(request, student);
+			if (course == null)
+			{
+				throw new ArgumentNullException();
+			}
+			await _unit.SaveChangesAsync();
+			var level = await _unit.Level.FetchLevelAsync(course.LevelId);
```

Comment:
why are you fetching this here..

Reply:
to return a response with levelName when addingCourse